dbt_common/context.py

```diff
@@ -1,20 +1,29 @@
 from contextvars import ContextVar, copy_context
 from typing import List, Mapping, Optional
 
-from dbt_common.constants import SECRET_ENV_PREFIX
+from dbt_common.constants import PRIVATE_ENV_PREFIX, SECRET_ENV_PREFIX
 
 
 class InvocationContext:
     def __init__(self, env: Mapping[str, str]):
-        self._env = env
+        self._env = {k: v for k, v in env.items() if not k.startswith(PRIVATE_ENV_PREFIX)}
         self._env_secrets: Optional[List[str]] = None
+        self._env_private = {
+            k[len(PRIVATE_ENV_PREFIX) :]: v
```

Comment:
what's the motivation for stripping the prefix off of private envs?

Reply:
My take is that it makes them less annoying and verbose to look up later, but I could go either way.

Reply:
I think even the need to document it tilts me to the side of just keeping the prefix, so I'll just do it that way.

Reply:
Done!